source/scripting_v3/GTA/InteriorProxy.cs

```diff
@@ -0,0 +1,234 @@
+		/// </summary>
+		/// <param name="handle">The interior proxy handle.</param>
+		/// <returns>
+		/// Returns a <see cref="InteriorProxy"/> if this handle corresponds to a <see cref="InteriorProxy"/>.
+		/// Returns <see langword="null" /> if no <see cref="Entity"/> exists this the specified <paramref name="handle"/>
+		/// </returns>
+		public static InteriorProxy FromHandle(int handle) => SHVDN.NativeMemory.InteriorProxyHandleExists(handle) ? new InteriorProxy(handle) : null;
+
+		/// <summary>
+		/// The handle of this <see cref="InteriorProxy"/>.
+		/// </summary>
+		public int Handle
+		{
+			get; private set;
+		}
+
+		/// <summary>
+		/// The handle of this <see cref="InteriorProxy"/> translated to a native value.
+		/// </summary>
+		public ulong NativeValue
+		{
+			get => (ulong)Handle;
+			set => Handle = unchecked((int)value);
+		}
+
+		/// <summary>
+		/// Gets the memory address where the <see cref="InteriorProxy"/> is stored in memory.
+		/// </summary>
+		public IntPtr MemoryAddress => SHVDN.NativeMemory.GetInteriorProxyAddress(Handle);
+
+		/// <summary>
+		/// Gets or sets the position of this <see cref="AnimatedBuilding"/>.
+		/// </summary>
+		/// <value>
+		/// The position in world space.
+		/// </value>
+		public Vector3 Position
+		{
+			get
+			{
+				var address = MemoryAddress;
+				if (address == IntPtr.Zero)
+					return Vector3.Zero;
+
+				return new Vector3(SHVDN.NativeMemory.ReadVector3(address + 0x70));
+			}
+		}
+
+		/// <summary>
+		/// Gets the current <see cref="InteriorInstance"/> this <see cref="InteriorProxy"/> is using.
+		/// </summary>
+		/// <remarks>returns <see langword="null" /> if this <see cref="InteriorProxy"/> is not using any <see cref="InteriorInstance"/>.</remarks>
+		public InteriorInstance CurrentInteriorInstance
+		{
+			get
+			{
+				var address = MemoryAddress;
+				if (address == IntPtr.Zero)
+					return null;
+
+				var interiorInstHandle = SHVDN.NativeMemory.GetAssociatedInteriorInstHandleFromInteriorProxy(Handle);
+				return interiorInstHandle != 0 ? new InteriorInstance(interiorInstHandle) : null;
+			}
+		}
+
+		/// <summary>
+		/// Gets the model this <see cref="InteriorProxy"/> will load.
+		/// </summary>
+		public Model InteriorModel
+		{
+			get
+			{
+				var address = MemoryAddress;
+				if (address == IntPtr.Zero)
+					return null;
+
+				return new Model(SHVDN.NativeMemory.ReadInt32(address + 0xE4));
+			}
+		}
+
+		/// <summary>
+		/// Gets a value indicating whether this <see cref="InteriorProxy"/> will behave as if interior is not loaded completely.
+		/// Does not prevent from having a <see cref="InteriorInstance"/>.
+		/// </summary>
+		/// <value>
+		/// <see langword="true" /> if this <see cref="InteriorProxy"/> will behave as if interior is not loaded completely; otherwise, <see langword="false" />.
+		/// </value>
+		public bool IsInteriorDisabled => Function.Call<bool>(Hash.IS_INTERIOR_DISABLED, Handle);
```

Comment:
I think all of these properties would be better without "Interior" in the name (since the class is already named as such), similar to properties in the other entity classes (e.g. `Vehicle.IsStolen` instead of `Vehicle.IsVehicleStolen`, even though native name is `IS_VEHICLE_STOLEN`). So in this case `IsDisabled` for example, instead of `IsInteriorDisabled`.